### PR TITLE
Add 'namespace' key to json to allow custom namespaces and upgrade the cmake

### PR DESCRIPTION
**Pull request by NicolasIRAGNE**

### CODE REVIEW

File: src/EnumGenerator.py [hunk 1]
```diff
@@ -31,6 +31,14 @@ def main():
...
     enumName = os.path.basename(jsonFile).rsplit('.', 1)[0]
     enumValuesArray = enumJson['values']
+    enumNamespace = ''
+    enumNamespaceStart = ''
+    enumNamespaceEnd = ''
+    if "namespace" in enumJson:
+        enumNamespace = enumJson['namespace']
+        enumNamespaceStart = "namespace " + enumNamespace + " {"
+        enumNamespaceEnd = "} // namespace " + enumNamespace
+        enumNamespace = enumNamespace + "::"
```
Comment (iburel) [inline src/EnumGenerator.py:41]: You can do `enumNamespace = enumJson['namespace'] + '::'` directly

File: src/EnumGenerator.py [hunk 2]
```diff
@@ -31,6 +31,14 @@ def main():
 
     enumName = os.path.basename(jsonFile).rsplit('.', 1)[0]
     enumValuesArray = enumJson['values']
+    enumNamespace = ''
+    enumNamespaceStart = ''
+    enumNamespaceEnd = ''
+    if "namespace" in enumJson:
+        enumNamespace = enumJson['namespace']
+        enumNamespaceStart = "namespace " + enumNamespace + " {"
```
Comment (iburel) [inline src/EnumGenerator.py:39]: Use single quotes

File: src/EnumGenerator.py (hunk 1 above)
Comment (iburel) [inline src/EnumGenerator.py:41]: You can name it enumNamespacePrefix instead

**Comment by NicolasIRAGNE** [comment]
Should be fixed now. Sorry for the late fixes, I was at McDonald's.

**Pull request merged**